lib/phoenix_live_view/utils.ex

```diff
@@ -28,21 +28,21 @@ defmodule Phoenix.LiveView.Utils do
   @doc """
   Configures the socket for use.
   """
-  def configure_socket(%{id: nil, assigns: assigns, view: view} = socket, private, action) do
+  def configure_socket(%{id: nil, assigns: assigns, view: view} = socket, private, action, flash) do
     %{
       socket
       | id: random_id(),
         private: private,
-        assigns: configure_assigns(assigns, view, action)
+        assigns: configure_assigns(assigns, view, action, flash)
     }
   end
 
-  def configure_socket(%{assigns: assigns, view: view} = socket, private, action) do
-    %{socket | private: private, assigns: configure_assigns(assigns, view, action)}
+  def configure_socket(%{assigns: assigns, view: view} = socket, private, action, flash) do
+    %{socket | private: private, assigns: configure_assigns(assigns, view, action, flash)}
   end
 
-  defp configure_assigns(assigns, view, action) do
-    assigns |> Map.put(:live_view_module, view) |> Map.put(:live_view_action, action)
+  defp configure_assigns(assigns, view, action, flash) do
+    Map.merge(assigns, %{live_view_module: view, live_view_action: action, flash: flash || %{}})
```

Comment:
Maybe flash can be `nil` because of the `verify` return. I would rather assume it is always a map at this point and remove the amount of "unknown" data in the system.